### PR TITLE
feat(api): add copy operation

### PR DESCRIPTION
**Pull request by Dyostiq**
### Overview

_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

_Please explain how to test the PR: ID of a dataset, steps to reach the feature, etc._

### Feature relevant tickets

_Link to the related task manager tickets_

---

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

### CODE REVIEW

**Comment by Dyostiq** [comment]
I know that tests fail :) but still the review is welcome

File: api/apps/api/src/modules/scenarios-features/copy/copy-data-provider.service.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+@Injectable()
+export class CopyDataProvider {
+  constructor(
+    @InjectEntityManager()
+    private readonly apiEntityManager: EntityManager,
+    private readonly projects: ProjectsCrudService,
+  ) {}
+
+  async prepareData(
```
Comment (kgajowy) [inline api/apps/api/src/modules/scenarios-features/copy/copy-data-provider.service.ts:18]: ~ `getProjectSetup` ?

Comment (kgajowy) [reply api/apps/api/src/modules/scenarios-features/copy/copy-data-provider.service.ts:18]: (as this piece will be most likely shared across all operations)

File: api/apps/api/src/modules/scenarios-features/copy/copy-operation.service.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  ) {}
+
+  async copy(command: CreateFeaturesCommand): Promise<{ id: string }[]> {
+    const {
+      project,
+      protectedAreaFilterByIds,
+      planningAreaLocation,
+    } = await this.copyDataProvider.prepareData(command);
+    const { parameters, query } = this.copyQuery.prepareStatement(
+      command,
```
Comment (kgajowy) [inline api/apps/api/src/modules/scenarios-features/copy/copy-operation.service.ts:25]: does it need the command itself?

**Review by kgajowy — APPROVED** [review]

File: api/apps/api/src/modules/scenarios-features/copy/copy-operation.service.ts (hunk 2 above)
Comment (Dyostiq) [reply api/apps/api/src/modules/scenarios-features/copy/copy-operation.service.ts:25]: the entire service could be in fact a command handler

**Comment by hotzevzl** [comment]
@Dyostiq structure and strategy look ok to me.

Since this whole über-story is a fairly tricky one, I would suggest to log queries and parameters in this phase so we can keep an eye on what gets sent to db and we can more easily debug things, and we can then fully remove in a second pass (or just rely on log levels to omit these in production)

**Pull request merged**